contents/dotfiles/Dotfiles.md

```diff
@@ -0,0 +1,233 @@
+
+<pic src="syntax-highlighting.gif" alt="Git flag autocomplete" width="60%">
+
+  <sub> _Figure 6. Syntax Highlighting [(Tutorial)](https://github.com/zsh-users/zsh-syntax-highlighting)_ </sub>
+
+</pic>
+
+### Getting Started
+
+While dotfiles are easy to setup and configure, they can become disorganised over time as a developer as more and more configurations are added. As much as possible, they should be managed like any other software engineering project, applying software design principles such as modularisation to make them extensible and future-proof.
+
+#### Management Strategies
+
+There are many stratgies for managing dotfiles, but virtually all of them revolve around storing them in `git` repositories. The nature of dotfiles make `git` and ideal management tool - they continously evolve over time as the user adds more configurations, and it may be useful to track old dotfiles for future reference.
+
+Another major benefit of managing dotfiles with `git` is that they can then be pushed to online repositories like GitHub or GitLab. These dotfiles can then be easily reused on other systems by simply pulling from these online repositories, allowing users to port their painstakingly created dotfiles anywhere. Since certain dotfiles may also vary across different systems (e.g. bash configurations may be different), users can use different branches to differentiate between these systems. It almost seems as though `git` was created for managing dotfiles!
+
+The main difference between different strategies are how these dotfiles should be linked from the git repository into the system, since dotfiles can exist in different directories. The 3 most commonly used strategies are detailed below:
+
+1. **Use a [git worktree](https://www.atlassian.com/git/tutorials/dotfiles):**\
+The strategy uses a less well-known Git functionality of storing the Git worktree separately from the Git directory. This consists of a Git bare repository in a "side" folder (like `$HOME/.cfg` or `$HOME/.myconfig`) using a specially crafted alias so that commands are run against that repository and not the usual .git local folder, which would interfere with any other Git repositories around.
+2. **Use [symlinking](https://opensource.com/article/19/3/move-your-dotfiles-version-control):**\
+This strategy involves using symlinks (The `ln` command) to create a shortcut linking to the dotfiles repository. The `ln` command is used to create links in your (unix-based) system. The syntax is `ln -s <actual location of the file> <name and location you want to see that file under>`. For example, the command below will result in the gitconfig in the dotfiles directory to be accessible from the `~/.gitconfig` location, which is where Git is expecting to see all the Git preferences set.
+
+```bash
+ln -s ~/dotfiles/gitconfig ~/.gitconfig
+```
```

Comment:
Indent this code block so that it appears under the bullet point.
Markbind support syntax highlighting of inline code too. e.g.,

```
2. **Use [symlinking](https://opensource.com/article/19/3/move-your-dotfiles-version-control):**\
This strategy ...<br>
   `ln -s ~/dotfiles/gitconfig ~/.gitconfig`{.bash}
```